src/main/java/org/sagebionetworks/web/client/widget/ReactComponentV2.java

```diff
@@ -186,21 +178,23 @@ public HandlerRegistration addClickHandler(ClickHandler handler) {
   }
 
   public void render() {
-    if (isRenderedAsReactComponentChild()) {
-      // This component will be rendered as a child of another React component, so destroy the root if one exists
-      destroyRoot();
-    }
+    // This component will be rendered as a child of another React component, so destroy the root if one exists
+    boolean shouldDestroyRoot = isRenderedAsReactComponentChild();
 
     // This component may be a React child of another component, so retrieve the root widget that renders this component tree.
     ReactComponentV2<?, ?> componentToRender = getRootReactComponentWidget();
 
-    // Asynchronously schedule creating a root in case we queued up an unmount of the current root
+    // Asynchronously schedule root operations in case the component is in the middle of an asynchronous render cycle
     // See https://stackoverflow.com/questions/73459382
     Timer t = new Timer() {
       @Override
       public void run() {
-        componentToRender.createRoot();
+        if (shouldDestroyRoot) {
+          destroyRoot();
+        }
+
         // Create a fresh ReactElement tree and render it
+        componentToRender.createRoot();
         componentToRender.root.render(componentToRender.createReactElement());
       }
     };
```

Comment:
Combine the destroyRoot/createRoot into one sequential operation to handle race conditions where appended GWT widgets could not be updated.